src/dev/run_quick_checks.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { exec } from 'child_process';
+import { cpus } from 'os';
+import { join, isAbsolute } from 'path';
+import { readdirSync, readFileSync } from 'fs';
+
+import { run, RunOptions } from '@kbn/dev-cli-runner';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { ToolingLog } from '@kbn/tooling-log';
+
+const MAX_PARALLELISM = cpus().length - 1;
```

Comment:
From the Node docs:

os.cpus().length should not be used to calculate the amount of parallelism available to an application. Use [os.availableParallelism()](https://nodejs.org/docs/latest-v20.x/api/os.html#osavailableparallelism) for this purpose.

Reply:
good catch